pgjdbc/src/main/java/org/postgresql/jdbc/PgResultSet.java

```diff
@@ -280,16 +281,22 @@ public ResultSetMetaData getMetaData() throws SQLException {
           // we have updatable cursors it must be readonly.
           ResultSet rs =
               connection.execSQLQuery(sb.toString(), resultsettype, ResultSet.CONCUR_READ_ONLY);
-          //
-          // In long running transactions these backend cursors take up memory space
-          // we could close in rs.close(), but if the transaction is closed before the result set,
-          // then
-          // the cursor no longer exists
 
-          sb.setLength(0);
-          sb.append("CLOSE ");
-          Utils.escapeIdentifier(sb, cursorName);
-          connection.execSQLUpdate(sb.toString());
+          /*
+          If the user has set a fetch size we can't close the cursor yet.
+          Issue https://github.com/pgjdbc/pgjdbc/issues/2227
+           */
+          if (connection.getDefaultFetchSize() == 0 ) {
```

Comment:
Should we check cursor's property rather than the one on the connection?

Reply:
Does that require a round trip ?

Reply:
It doe not, however, it looks strange to use connection-related info (which might be changed over time).

Reply:
Honestly I don't know where to find this. The connection was the first place I looked

Reply:
@vlsi  better ?